clients/drcachesim/common/trace_entry.cpp

```diff
@@ -88,5 +88,55 @@ const char *const trace_type_names[] = {
     "untaken_jump",
 };
 
+const char *const trace_version_names[] = {
+    "<unknown>",
+    "<unknown>",
+    "trace_entry_version_no_kernel_pc",
+    "trace_entry_version_kernel_pc",
+    "trace_entry_version_encodings",
+    "trace_entry_version_branch_info",
+    "trace_entry_version_frequent_timestamps",
+};
+
+const char *const trace_marker_names[] = {
```

Comment:
Would we want the strings at clients/drcachesim/tools/view.cpp to be replaced by these? Maybe. Add a TODO at view.cpp.

Reply:
That would change all the output recorded in the docs here and in the sample traces repo and in internal docs so would not be taken lightly.

Reply:
The reverse might be an easier thing to do.  See all the comments about sharing code: eliminate this array and use the existing public pretty-printer's marker printing code.

Reply:
https://github.com/DynamoRIO/dynamorio/pull/6771#discussion_r1571292467

Reply:
> The reverse might be an easier thing to do. See all the comments about sharing code: eliminate this array and use the existing public pretty-printer's marker printing code.

I modified this array to be coherent with view.cpp (which, now, it also uses).
I think we want a simple string representation of a marker type.
What view.cpp does is merging together the string representation of a marker type + a marker value.

Now, `trace_marker_names[]` contains the string representation of marker types, and it's used in the new `trace_marker_type_value_as_string()`, which returns a string representation of marker type + value (and it's in common between view.cpp and record_view.cpp).